lib/components/form/form_field/gf_form_field.dart

```diff
@@ -0,0 +1,519 @@
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    return  getFormWidget(widget.gfFormFieldType);}
+
+
+ Widget getFormWidget(GfFormFieldType type){
+   Widget retwidget=Container();
+   switch(type){
+     case GfFormFieldType.name:
+       retwidget=  Container(
+           margin: EdgeInsets.symmetric(vertical: widget.marginvertical??2,
+               horizontal: widget.marginhorizontal??2),
+           padding: EdgeInsets.symmetric(vertical: widget.paddingvertical??2,
+               horizontal: widget.paddinghorizontal??2),
+           child:
+           GFTextField(
+               decoration:
+                GfFormFieldDecoration(
+                    context: context,
+                    shape: widget.shape??GFTextFieldShape.roundedsquare,
+                    borderRadius: widget.borderradius??6.0,
+                    editingBorderColor: widget.editingbordercolor,
+                    idleBorderColor: widget.idlebordercolor,
+                    gfprefixIcon: widget.iconPrefix,
+                    bgfilled: true,
+                    bgcolor: widget.backgroundcolor,
+                    hinttext: widget.hintText.isNotEmpty?widget.hintText:'Name',
+                ),
+               controller: widget.controller,
+               fieldinitialValue: widget.initialValue,
+               focusNode: widget.focusNode,
+               keyboardType: TextInputType.name,
+               textCapitalization: TextCapitalization.words,
+               textInputAction: widget.textInputAction,
+               style: widget.style,
+               strutStyle: widget.strutStyle,
+               textDirection: widget.textDirection,
+               textAlign: widget.textAlign,
+               textAlignVertical: widget.textAlignVertical,
+               autofocus: widget.autofocus,
+               readOnly: widget.readOnly,
+               toolbarOptions: widget.toolbarOptions,
+               showCursor: widget.showCursor,
+               obscuringCharacter: widget.obscuringCharacter,
+               obscureText: widget.obscureText,
+               autocorrect: widget.autocorrect,
+               smartDashesType: widget.smartDashesType,
+               smartQuotesType: widget.smartQuotesType,
+               enableSuggestions: widget.enableSuggestions,
+               autovalidate: widget.autovalidate,
+               maxLengthEnforced: widget.maxLengthEnforced,
+               maxLines: widget.maxLines,
+               minLines: widget.minLines,
+               expands: widget.expands,
+               maxLength: widget.maxLength,
+               onChanged: widget.onChanged,
+               onTap: widget.onTap,
+               onEditingComplete: widget.onEditingComplete,
+               onFieldSubmitted: widget.onFieldSubmitted,
+               onfieldSaved: widget.onSaved,
+               validatornew: (value) {
+                 if (value == null || value.isEmpty) {
+                   return 'Please enter name';
```

Comment:
we need to think about localizations. not every app will only be available in english, so we need to provide a way that user can set the text

Reply:
Okay I will make changes